src/Materials/Textures/multiRenderTarget.ts

```diff
@@ -241,17 +241,6 @@ export class MultiRenderTarget extends RenderTargetTexture {
      * @param index The index of the texture to replace
      */
     public setInternalTexture(texture: InternalTexture, index: number, attachmentIndex: number = 0, disposePrevious: boolean = true) {
-        // if (index === 0) {
-        //     // This function is just meant to replace textures, not the framebuffer
-        //     // So we keep references
-        //     texture._framebuffer = this._texture!._framebuffer;
-        //     // Prevents the dispose of the framebuffer
-        //     this._texture!._framebuffer = null;
-        //     texture._MSAAFramebuffer = this._texture!._MSAAFramebuffer;
-        //     this._texture!._MSAAFramebuffer = null;
-        //     this._texture = texture;
-        // }
-
         if (disposePrevious && this._internalTextures[index] && index !== 0) {

```

Comment:
What about having a method like this in `engine.multiRender.ts`:
```typescript
ThinEngine.prototype.updateMultipleRenderTargetTexture = function(textures: InternalTexture[], texture: InternalTexture, index: number, attachmentIndex: number = 0, disposePrevious: boolean = true): void {
    if (disposePrevious && textures[index] && index !== 0) {
        textures[index].dispose();    
    }
    
    textures[index] = texture;

    this.bindTextureFramebuffer(textures[0]._framebuffer!, texture, attachmentIndex);
}
```
Then:
```typescript
public setInternalTexture(texture: InternalTexture, index: number, attachmentIndex: number = 0, disposePrevious: boolean = true) {
    this._engine.updateMultipleRenderTargetTexture (this._internalTextures, texture, attachmentIndex, disposePrevious);

    if (!this.textures[index]) {
        this.textures[index] = new Texture(null, this.getScene());
    }
    this.textures[index]._texture = this._internalTextures[index];

    this._count = this._internalTextures.length;
    // TODO : update types and samplingModes
}
```
That way you can remove the `_frameBuffer` data member, as well as `_getFrameBuffer()`.

Also, why do we need an `attachmentIndex` that would be different from `index`? We can't simply use `index` instead of `attachmentIndex`?